src/Microsoft.AspNet.Mvc.Core/Controllers/FilterActionInvoker.cs

```diff
@@ -237,7 +242,33 @@ private IFilterMetadata[] GetFilters()
                 _filterProviders[i].OnProvidersExecuted(context);
             }
 
-            return context.Results.Select(item => item.Filter).Where(filter => filter != null).ToArray();
+            var count = 0;
+            for (var i = 0; i < items.Count; i++)
```

Comment:
Counting up front to avoid resize when using a list, and to avoid the need to call `ToArray`